main.tf

```diff
@@ -10,8 +10,8 @@ resource "azurerm_machine_learning_workspace" "this" {
   location                = var.location
   resource_group_name     = var.resource_group_name
   application_insights_id = azurerm_application_insights.this.id
-  key_vault_id            = var.key_vault_id
-  storage_account_id      = var.storage_account_id
+  key_vault_id            = var.key_vault == null ? module.avm_res_keyvault_vault[0].resource_id : var.key_vault.id 
```

Comment:
if the resource itself uses resource_id it is better for you to also use resource_id and not to id so if somoene created the resource outside they could just pass in the whole resource and it should work.